compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.h

```diff
@@ -17,6 +17,19 @@
 
 namespace mlir::iree_compiler {
 
+//===----------------------------------------------------------------------===//
+// Constants
+//===----------------------------------------------------------------------===//
+
+/// Named attributes used in the `translation_info`'s config dictionary
+/// attribute. These are used to override default pass heuristics at the
+/// function granularity.
+namespace LLVMGPUAttrNames {
+inline constexpr StringLiteral kNoReorderWorkgroups = "no_reorder_workgroups";
+inline constexpr StringLiteral kNoReduceSharedMemoryBankConflicts =
+    "no_reduce_shared_memory_bank_conflicts";
```

Comment:
I don't know if there is a preference of making attributes enable things (reorder_workgroups, reduce_shared_memory_bank_conflicts) vs disable things. This looks fine, maybe we can revisit it in future.

Reply:
We recently start using attributes to determine whether we want peeling or not in CPU backends. I think having such support is okay because it also exposes the control to `KernelConfig.cpp`.

Reply:
Because these override global defaults, we may need either depending on the global settings. In this case these optimizations are enabled globally, so I need knobs to disable them. We can add the duals that enable them later when we need that.